docs/reconciler-proposal/kris-1/interface.go

```diff
@@ -0,0 +1,62 @@
+// Copyright © 2017 The Kubicorn Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloud
+
+import (
+	"github.com/kris-nova/kubicorn/apis/cluster"
+)
+
+// Model will model resources to an API representation of infrastructure
+type Model interface {
+
+	// Get will return the map of ordered resources against a cloud
+	Get() map[int]Resource
```

Comment:
Why a map if they are ordered? Why not a slice? Do we expect iteration across the objects to pretty much always happen in order?